userspace/libsinsp/cri.hpp

```diff
@@ -656,85 +694,38 @@ inline std::string cri_interface<api>::get_container_image_id(const std::string
 	return "";
 }
 
-template<typename api>
-inline bool cri_interface<api>::parse_containerd(const typename api::ContainerStatusResponse &status,
```

Comment:
Removing `parse_containerd` brings significant benefits by enhancing the clarity and transparency of the codebase. This is because the parsing previously done here always also applied for `cri-o` and therefore at best this approach was confusing overall.

Now the main `parse` function calls all parsers and we get rid of this nested additional jump.

Also I have the suspicion we were inefficient with the IP enrichment / lookup before (it may have been called twice?), now the main`parse` function has a very clear structure ensuring we make only one call for container and pod sandbox status responses and parse them right away.